tests/test_types.py

```diff
@@ -488,6 +488,36 @@ class Model(BaseModel):
     assert Model().foo is None
 
 
+def test_constrained_frozenset():
+    class Model(BaseModel):
+        x: FrozenSet[str] = Field(..., min_items=1, max_items=3)
+
+    with pytest.raises(ValidationError) as exc_info:
+        Model(x=frozenset())
+    assert exc_info.value.errors() == [
+        {
+            'loc': ('x',),
+            'msg': 'ensure this value has at least 1 items',
+            'type': 'value_error.set.min_items',
+            'ctx': {'limit_value': 1},
+        }
+    ]
+
+    Model(x=frozenset({'a'}))
+    Model(x=frozenset({'a', 'b', 'c'}))
```

Comment:
can we check that `.dict()` is correct.

Can we also try with a list or normal set or something?